code/Model/SiteTree.php

```diff
@@ -2711,6 +2712,43 @@ public function getPageLevel() {
 	}
 
 	/**
+	 * @param string $action
+	 * @return Controller
+	 */
+	public function getAssociatedController($action = null) {
```

Comment:
Hmm, sounds interesting. Would we run the risk of ending up with circular dependencies, though? Page has a controller, which has a page, which has a controller... you get the idea.